### ISSUE TITLE
What happens if HttpAuthenticationMechanism and <login-config> both specified?

### ISSUE THREAD

**Issue by wmhopkins** [javaee/security-spec#45]
The spec currently says, "If an application provides an HttpAuthenticationMechanism and also configures a login-config element in web.xml, the container SHOULD fail deployment, but MAY choose to proceed, using only the HttpAuthenticationMechanism (i.e., ignoring the login-config from web.xml)."

This seems reasonable, but as a practical matter, will it ever be possible for a servlet container to reliably do this (considering that the HAM can be any scope)? If not, should the language change ("MAY fail deployment or choose to proceed", "... result is undefined")?

This is a similar issue to #43,

**Comment by ggam** [comment]
I don't think silently ignoring one login config is a good idea. You can have a library which contains a `web-fragment.xml` that specifies `login-config` you're not aware of, it's not only about the top-level `web.xml` (although I admit the case I describe would be rather uncommon).

I prefer to leave it as is, but I don't know the technical implications it may have. 

As a side note, what happens now if a `login-config` is defined and then a `ServerAuthModule` is registered (application or server wide)? Does it ignore `login-config`?

**Comment by arjantijms** [comment]
>As a side note, what happens now if a login-config is defined and then a ServerAuthModule is registered (application or server wide)? Does it ignore login-config?

yes, this is indeed what happens now and is how the JASPIC spec defined it.

All containers basically do:

```java
if (jsr196Configured {
   // get sam
   // invoke sam
} else {
  // proceed with proprietary
}
```

Internally (container specific) it should be possible to detect whether a `login-config` was configured. The scope of the `HttpAuthenticationMechanism` does not matter (you only have to detect that there's an enabled Bean, which you can do very early as CDI boots).

**Comment by wmhopkins** [comment]
The team doing the GF integration seemed to think detecting this at deployment time was difficult, because CDI wasn't initialized at that time. You could write code that would run once CDI is initialized to go find the descriptor and look at it, but it might or might not be easy to get the descriptor from a CDI bean.

I think all we need to do here is change the language to say the server MAY detect this and fail deployment, but is not required to. If it proceeds, it MUST use the HAM instead of the <login-config>. As @arjantijms points out, it already works that way for JASPIC, which means it works that way for HAM.

**Comment by wmhopkins** [comment]
Fixed by #58 .
